src/tic_tac_toe/core.clj

```diff
@@ -0,0 +1,34 @@
+(ns tic-tac-toe.core
+  (:gen-class)
+  (:use [clojure.math.numeric-tower :only [expt]]
+        [tic-tac-toe.board :as board]
+        [tic-tac-toe.player :as player]
+        [tic-tac-toe.user-interface :as ui]))
+
+(def size 3)
+
+(defn game-setup
+  []
+  (hash-map :size size :board (board/make size)))
+
+(defn take-turn
+  [board size]
+  (do (ui/display-board board 3)
```

Comment:
I tend to see the use of `do` as a smell. I imagine you are using `do` to get around the fact the `ui/display-board` has a `nil` return value?

The idiomatic approach is to do something like

```clojure
(ui/message (ui/request-move (ui-display-board state)))
```

Where a single data structure is passed through and potentially modified in the process. This would also enable you to use the [threading macro](https://clojure.org/guides/threading_macros) which significantly improves the readability (IMO)

Reply:
Started a [threading macro](https://github.com/saralein/tic-tac-toe_clojure/pull/1/files#diff-57905e40aaa721e61a7ea7ef112d9ed3R14).  I still haven't been able to figure out how to remove the do entirely (here and a few other places).  I'll probably ask for some advice later today.

Reply:
Added some [threading](https://github.com/saralein/tic-tac-toe_clojure/pull/1/files#diff-52c69c82bcdbcc16e3156d4a9033a37cR29) to UI protocol and to the [game](https://github.com/saralein/tic-tac-toe_clojure/pull/1/files#diff-1e397d0c219d54561be23ddb30fd0516R12).